src/vm_stack.zig

```diff
@@ -976,7 +976,8 @@ const InstructionFuncs = struct {
             }
 
             const mem = buffer[offset..end];
-            const value = std.mem.readIntSliceLittle(read_type, mem);
+            const byte_count = bit_count / 8;
+            const value = std.mem.readInt(read_type, mem[0..byte_count], .little);
```

Comment:
I think this could just be `std.mem.readInt(read_type, mem, .little)`? I don't quite see the purpose of the `[0..byte_count]` part.

Reply:
std.mem.readInt takes an array, so the slice needs to be comptime known. Currently `mem` here is a runtime slice, so the extra `[0..byte_count]` part is used to create the comptime known array length.

Reply:
Ah I see, that does make sense.